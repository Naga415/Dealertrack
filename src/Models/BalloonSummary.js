'use strict';
export default class BalloonSummary {
  constructor() {
    this.payment = null;
    this.amountFinananced = null;
    this.holdDeposit = null;
    this.purchasePriceWithAdds = null;
    this.discountFromSellingPrice = null;
    this.netTrade = null;
    this.balloonAmount = null;
    this.interestRateTotal = null;
    this.totalPayments = null;
    this.subtotal = null;
  }
}
