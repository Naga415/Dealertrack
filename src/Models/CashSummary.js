'use strict';
export default class CashSummary {
  constructor() {
    this.balanceDue = null;
    this.frontGross = null;
    this.backGross = null;
    this.holdDeposit = null;
    this.purchasePriceWithAdds = null;
    this.discountFromSellingPrice = null;
    this.netTrade = null;
    this.subtotal = null;
  }
}
