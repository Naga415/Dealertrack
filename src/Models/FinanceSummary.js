'use strict';
export default class FinanceSummary {
  constructor() {
    this.payment = null;
    this.amountFinanced = null;
    this.amountDueAtSigning = null;
    this.balanceDue = null;
    this.holdDeposit = null;
    this.purchasePriceWithAdds = null;
    this.discountFromSellingPrice = null;
    this.netTrade = null;
    this.interestRateTotal = null;
    this.totalPayments = null;
    this.subtotal = null;
  }
}
