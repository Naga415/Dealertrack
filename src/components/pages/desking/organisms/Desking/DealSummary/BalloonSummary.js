'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Summary from '../../../molecules/Summary/Summary';
import BalloonSummaryModel from '../../../../../../models/BalloonSummary';

// eslint-disable-next-line react/prefer-stateless-function
class BalloonSummary extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <Summary title="Balloon Deal" htmlId="BalloonSummary">
        <Summary.Item label="Hold Deposit" value={this.props.financialSummary.holdDeposit}/>
        <Summary.Item label="Purchase Price w/ Adds" value={this.props.financialSummary.purchasePriceWithAdds}/>
        <Summary.Item label="Discount from Selling Price" value={this.props.financialSummary.discountFromSellingPrice}/>
        <Summary.Item label="Net Trade" value={this.props.financialSummary.netTrade}/>
        <Summary.Item label="Balloon Amount" value={this.props.financialSummary.balloonAmount}/>
        <Summary.Item label="Interest Rate Total" value={this.props.financialSummary.interestRateTotal}/>
        <Summary.Item label="Total of Payments" value={this.props.financialSummary.totalPayments}/>
      </Summary>
    );
  }
}

BalloonSummary.propTypes = {
  financialSummary: PropTypes.instanceOf(BalloonSummaryModel)
};

export default BalloonSummary;
