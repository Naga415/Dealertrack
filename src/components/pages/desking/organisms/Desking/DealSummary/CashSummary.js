'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Summary from '../../../molecules/Summary/Summary';
import CashSummaryModel from '../../../../../../models/CashSummary';

// eslint-disable-next-line react/prefer-stateless-function
class CashSummary extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <Summary title="Cash Deal" htmlId="CashSummary">
        <Summary.Item label="Hold Deposit" value={this.props.financialSummary.holdDeposit}/>
        <Summary.Item label="Purchase Price w/ Adds" value={this.props.financialSummary.purchasePriceWithAdds}/>
        <Summary.Item label="Discount from Selling Price" value={this.props.financialSummary.discountFromSellingPrice}/>
        <Summary.Item label="Net Trade" value={this.props.financialSummary.netTrade}/>
        <Summary.Item label="Interest Rate Total" value={this.props.financialSummary.interestRateTotal}/>
        <Summary.Item label="Total of Payments" value={this.props.financialSummary.totalPayments}/>
      </Summary>
    );
  }
}

CashSummary.propTypes = {
  financialSummary: PropTypes.instanceOf(CashSummaryModel)
};

export default CashSummary;
