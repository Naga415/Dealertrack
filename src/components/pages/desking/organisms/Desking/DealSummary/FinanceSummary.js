'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Summary from '../../../molecules/Summary/Summary';
import FinanceSummaryModel from '../../../../../../models/FinanceSummary';

// eslint-disable-next-line react/prefer-stateless-function
class FinanceSummary extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <Summary title="Finance Deal" htmlId="FinanceSummary">
        <Summary.Item label="Hold Deposit" value={this.props.financialSummary.holdDeposit}/>
        <Summary.Item label="Purchase Price w/ Adds" value={this.props.financialSummary.purchasePriceWithAdds}/>
        <Summary.Item label="Discount from Selling Price" value={this.props.financialSummary.discountFromSellingPrice}/>
        <Summary.Item label="Net Trade" value={this.props.financialSummary.netTrade}/>
        <Summary.Item label="Interest Rate Total" value={this.props.financialSummary.interestRateTotal}/>
        <Summary.Item label="Total of Payments" value={this.props.financialSummary.totalPayments}/>
      </Summary>
    );
  }
}

FinanceSummary.propTypes = {
  financialSummary: PropTypes.instanceOf(FinanceSummaryModel)
};

export default FinanceSummary;

