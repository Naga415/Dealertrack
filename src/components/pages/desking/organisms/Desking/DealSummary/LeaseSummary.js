'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Summary from '../../../molecules/Summary/Summary';
import LeaseSummaryModel from '../../../../../../models/LeaseSummary';

// eslint-disable-next-line react/prefer-stateless-function
class LeaseSummary extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <Summary title="Lease Deal" htmlId="LeaseSummary">
        <Summary.Item label="Hold Deposit" value={this.props.financialSummary.holdDeposit}/>
        <Summary.Item label="Purchase Price w/ Adds" value={this.props.financialSummary.purchasePriceWithAdds}/>
        <Summary.Item label="Discount from Selling Price" value={this.props.financialSummary.discountFromSellingPrice}/>
        <Summary.Item label="Net Trade" value={this.props.financialSummary.netTrade}/>
        <Summary.Item label="Depreciation" value={this.props.financialSummary.depreciation}/>
        <Summary.Item label="Rent Charge Total" value={this.props.financialSummary.rentChargeTotal}/>
        <Summary.Item label="Base Monthly Payment" value={this.props.financialSummary.baseMonthlyPayment}/>
        <Summary.Item label="Days to 1st Payment" value={this.props.financialSummary.daysToFirstPayment}/>
        <Summary.Item label="Total Interest Paid" value={this.props.financialSummary.totalInterestPaid}/>
        <Summary.Item label="Total of Payments" value={this.props.financialSummary.totalOfPayments}/>
      </Summary>
    );
  }
}

LeaseSummary.propTypes = {
  financialSummary: PropTypes.instanceOf(LeaseSummaryModel)
};

export default LeaseSummary;

