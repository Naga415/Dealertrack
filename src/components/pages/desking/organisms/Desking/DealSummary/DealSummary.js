'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import FinancialSummaryModel from '../../../../../../models/FinanceSummary';
import LeaseSummaryModel from '../../../../../../models/LeaseSummary';
import CashSummaryModel from '../../../../../../models/CashSummary';
import BalloonSummaryModel from '../../../../../../models/BalloonSummary';
import FinanceSummary from './FinanceSummary';
import LeaseSummary from './LeaseSummary';
import CashSummary from './CashSummary';
import BalloonSummary from './BalloonSummary';

// eslint-disable-next-line react/prefer-stateless-function
export class DealSummaryClass extends React.PureComponent {
  render() {
    const summary = this.props.financialSummary;
    if (summary instanceof FinancialSummaryModel){
      return <FinanceSummary financialSummary={this.props.financialSummary}/>;
    }
    else if (summary instanceof LeaseSummaryModel){
      return <LeaseSummary financialSummary={this.props.financialSummary}/>;
    }
    else if (summary instanceof CashSummaryModel){
      return <CashSummary financialSummary={this.props.financialSummary}/>;
    }
    else if (summary instanceof BalloonSummaryModel){
      return <BalloonSummary financialSummary={this.props.financialSummary}/>;
    }
  }
}
DealSummaryClass.propTypes = {
  htmlId: PropTypes.string.isRequired,
  financialSummary: PropTypes.oneOfType([
    PropTypes.instanceOf(FinancialSummaryModel),
    PropTypes.instanceOf(LeaseSummaryModel),
    PropTypes.instanceOf(CashSummaryModel),
    PropTypes.instanceOf(BalloonSummaryModel)
  ])
};


export default DealSummaryClass;

