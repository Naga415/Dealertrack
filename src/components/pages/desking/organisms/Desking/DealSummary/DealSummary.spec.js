import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {DealSummaryClass as DealSummary} from './DealSummary';
import FinanceSummary from '../../../../../../models/FinanceSummary';
import CashSummary from '../../../../../../models/CashSummary';
import LeaseSummary from '../../../../../../models/LeaseSummary';
import BalloonSummary from '../../../../../../models/BalloonSummary';

describe('Deal Summary', () => {
  const financeSummary = new FinanceSummary();
  financeSummary.amountDueAtSigning = 2000;
  financeSummary.amountFinanced = 30000;
  financeSummary.payment = 550;
  financeSummary.discountFromSellingPrice = 4000;
  financeSummary.netTrade = 7000;
  financeSummary.interestRateTotal = 1500;
  financeSummary.totalPayments = 37000;

  const htmlId = "DealSummary";
  const financeProps = {
    htmlId,
    financialSummary: financeSummary
  };

  it('renders <DealSummary/> organism', () => {
    const component = renderer.create(<DealSummary {...financeProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders the correct amount for the summary items', () => {
    const component = mount(<DealSummary {...financeProps} />);
    expect(component.find('#NetTrade-value').text()).toEqual('$7,000');
    expect(component.find('#InterestRateTotal-value').text()).toEqual('$1,500');
    expect(component.find('#TotalofPayments-value').text()).toEqual('$37,000');
  });

  it('renders finance summary', () => {
    const component = mount(<DealSummary {...financeProps} />);
    expect(component.find('#FinanceSummary').length).toBe(1);
  });

  it('renders cash summary', () => {
    const financialSummary = new CashSummary();
    const props = {htmlId, financialSummary};
    const component = mount(<DealSummary {...props} />);
    expect(component.find('#CashSummary').length).toBe(1);
  });

  it('renders lease summary', () => {
    const financialSummary = new LeaseSummary();
    const props = {htmlId, financialSummary};
    const component = mount(<DealSummary {...props} />);
    expect(component.find('#LeaseSummary').length).toBe(1);
  });

  it('renders balloon summary', () => {
    const financialSummary = new BalloonSummary();
    const props = {htmlId, financialSummary};
    const component = mount(<DealSummary {...props} />);
    expect(component.find('#BalloonSummary').length).toBe(1);
  });
});

