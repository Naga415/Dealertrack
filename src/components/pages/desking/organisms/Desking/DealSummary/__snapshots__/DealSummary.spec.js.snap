// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal Summary renders <DealSummary/> organism 1`] = `
<div
  className="deal-summary"
  id="FinanceSummary"
>
  <div
    className="deal-summary__header"
  >
    <label
      htmlFor="FinanceSummaryHeader"
    >
      Finance Deal
    </label>
  </div>
  <div
    className="deal-summary__body container-fluid"
    id={undefined}
  >
    <div
      className="show-grid deal-summary-item row"
      id={undefined}
    >
      <div
        className="col-md-8"
        id={undefined}
      >
        <label
          className="deal-summary-item__label"
          htmlFor="Hold Deposit"
          id="HoldDeposit-label"
        >
          Hold Deposit
        </label>
      </div>
      <div
        className="col-md-4"
        id={undefined}
      >
        <label
          className="deal-summary-item__value"
          htmlFor="Hold Deposit"
          id="HoldDeposit-value"
        >
          _
        </label>
      </div>
    </div>
    <div
      className="show-grid deal-summary-item row"
      id={undefined}
    >
      <div
        className="col-md-8"
        id={undefined}
      >
        <label
          className="deal-summary-item__label"
          htmlFor="Purchase Price w/ Adds"
          id="PurchasePricew/Adds-label"
        >
          Purchase Price w/ Adds
        </label>
      </div>
      <div
        className="col-md-4"
        id={undefined}
      >
        <label
          className="deal-summary-item__value"
          htmlFor="Purchase Price w/ Adds"
          id="PurchasePricew/Adds-value"
        >
          _
        </label>
      </div>
    </div>
    <div
      className="show-grid deal-summary-item row"
      id={undefined}
    >
      <div
        className="col-md-8"
        id={undefined}
      >
        <label
          className="deal-summary-item__label"
          htmlFor="Discount from Selling Price"
          id="DiscountfromSellingPrice-label"
        >
          Discount from Selling Price
        </label>
      </div>
      <div
        className="col-md-4"
        id={undefined}
      >
        <label
          className="deal-summary-item__value"
          htmlFor="Discount from Selling Price"
          id="DiscountfromSellingPrice-value"
        >
          $4,000
        </label>
      </div>
    </div>
    <div
      className="show-grid deal-summary-item row"
      id={undefined}
    >
      <div
        className="col-md-8"
        id={undefined}
      >
        <label
          className="deal-summary-item__label"
          htmlFor="Net Trade"
          id="NetTrade-label"
        >
          Net Trade
        </label>
      </div>
      <div
        className="col-md-4"
        id={undefined}
      >
        <label
          className="deal-summary-item__value"
          htmlFor="Net Trade"
          id="NetTrade-value"
        >
          $7,000
        </label>
      </div>
    </div>
    <div
      className="show-grid deal-summary-item row"
      id={undefined}
    >
      <div
        className="col-md-8"
        id={undefined}
      >
        <label
          className="deal-summary-item__label"
          htmlFor="Interest Rate Total"
          id="InterestRateTotal-label"
        >
          Interest Rate Total
        </label>
      </div>
      <div
        className="col-md-4"
        id={undefined}
      >
        <label
          className="deal-summary-item__value"
          htmlFor="Interest Rate Total"
          id="InterestRateTotal-value"
        >
          $1,500
        </label>
      </div>
    </div>
    <div
      className="show-grid deal-summary-item row"
      id={undefined}
    >
      <div
        className="col-md-8"
        id={undefined}
      >
        <label
          className="deal-summary-item__label"
          htmlFor="Total of Payments"
          id="TotalofPayments-label"
        >
          Total of Payments
        </label>
      </div>
      <div
        className="col-md-4"
        id={undefined}
      >
        <label
          className="deal-summary-item__value"
          htmlFor="Total of Payments"
          id="TotalofPayments-value"
        >
          $37,000
        </label>
      </div>
    </div>
  </div>
</div>
`;
